lib/binding_of_caller/mri2.rb

```diff
@@ -17,25 +17,19 @@ def of_caller(n)
     # @return [Array<Binding>]
     def callers
       ary = []
-    
-      RubyVM::DebugInspector.open do |i|
-        n = 0
-        loop do
-          begin
-            b = i.frame_binding(n) 
-          rescue ArgumentError
-            break
-          end
 
+      RubyVM::DebugInspector.open do |dc|
+        locs = dc.backtrace_locations
+
+        locs.size.times do |i|
```

Comment:
Would you be able to explain a little more how this fix works? I don't doubt it does, but at first glance i can't see why this approach would work but the previous one would not -- as the previous one kept on iterating indefinitely until it encountered an error, so how could it end up with an empty array?